### PR TITLE
WEBAPP-644: EventJsonLd timezone is wrong

### PR DESCRIPTION
**Pull request by lulo4907**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

**Review by Schedulaar — COMMENTED** [review]
Well spotted, that this is caused by the plugin.
Indeed, I probably overlooked the usage in the events endpoint.
As we don't get a UTC normed date, we need to use timezone data for this.

Still, you can use MomentTimezoneDataPlugin to reduce the bundle size (including all timezone data would probably increase bundle size by ~200kB). What makes sense for us is to only include timezone data ranging from "this year two years onwards" or so. (see https://github.com/gilmoreorless/moment-timezone-data-webpack-plugin#all-zones-with-a-limited-date-range).
```js
const currentYear = new Date().getFullYear();
const plugin = new MomentTimezoneDataPlugin({
  startYear: currentYear,
  endYear: currentYear + 2
})
```
Additionally, one could think about limiting the data to only certain regions, but that's a bit risky...

Your build is currently failing, as your bundle exceeds the current budget of 1.5MiB. If your build is still failing with the optimizations applied above, you can think about increasing the budget.

**Comment by maxammann** [comment]
Thanks for your input @Schedulaar !
Yeah lets apply the time ranges and then increase the max. bundle size.

**Comment by lulo4907** [comment]
The plugin has been reentered, the entrypoint is now at 1.51 MiB.
What we need to discuss is the technical side of the time zones at the CMS. because not all selectable time zones are compatible with the plugin and runs into JS errors.

**Comment by maxammann** [comment]
@lulo4907 that is weird, can you explain and mention an example? The name should be the same as here: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

**Comment by maxammann** [comment]
To fix the build increase the asset size limit to 1.6 MiB

**Comment by maxammann** [comment]
Screenshot_1600074411


I noticed this on NATIVE. Could this be related?

**Review by Schedulaar — APPROVED** [review]

**Review by klinzo — APPROVED** [review]

File: tools/webpack.config.babel.js [hunk 1]
```diff
@@ -146,7 +148,7 @@ const createConfig = (env = {}) => {
         minimize: isProductionBuild
       }),
       // We only use moment-timezone for parsing GMT time-zoned data in the integreat-api-client
```
Comment (klinzo) [inline tools/webpack.config.babel.js:150]: Maybe update this comment accordingly

**Comment by lulo4907** [comment]
@maxammann 
The Webapp event example using "UTC-12" : ~/testumgebung/de/events/testevent-3
image

Depending on the configuration script more or less time zones are found. But some others, like "UTC-12" do not exist.

File: tools/webpack.config.babel.js [hunk 2]
```diff
@@ -147,7 +147,7 @@ const createConfig = (env = {}) => {
         debug: !isProductionBuild,
         minimize: isProductionBuild
       }),
-      // We only use moment-timezone for parsing GMT time-zoned data in the integreat-api-client
+      // We use moment-timezone for parsing a limited range of years here with GTM data in the integreat-api-client
```
Comment (Schedulaar) [inline tools/webpack.config.babel.js:150]:
```suggestion
      // As all parsed dates are in the near future, we limit the timezone-data to only include the next two years.
```

**Review by Schedulaar — APPROVED** [review]

**Review by maxammann — APPROVED** [review]

**Pull request merged**